File: CodeKata/Loop - 3/Python/187. Run-Length Encoding.py
v = input().strip()
if not v:
    print("")
else:
    result = ""
    count = 1
    for i in range(1, len(v)):
        if v[i] == v[i-1]:
            count += 1
        else:
            result += v[i-1] + str(count)
            count = 1
    result += v[-1] + str(count)
    print(result)

File: CodeKata/Loop - 3/Java Script/187. Run-Length Encoding.js
const readline = require("readline");

const inp = readline.createInterface({
    input: process.stdin
});

const userInput = [];

inp.on("line", (data) => {
    userInput.push(data);
});

inp.on("close", () => {

    let v = userInput[0].split("")

    if (!v) {
        console.log("");
    } else {
        let result = "";
        let count = 1;

        for (let i = 1; i < v.length; i++) {
            if (v[i] === v[i - 1]) {
                count++;
            } else {
                result += v[i - 1] + count;
                count = 1;
            }
        }
        result += v[v.length - 1] + count;

        console.log(result);
    }
});